src/main/java/mekanism/common/base/EnergyAcceptorTarget.java

```diff
@@ -0,0 +1,67 @@
+package mekanism.common.base;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Map.Entry;
+import net.minecraft.util.EnumFacing;
+
+public class EnergyAcceptorTarget {
+
+    private Map<EnumFacing, EnergyAcceptorWrapper> wrappers = new HashMap<>();
+    private Map<EnumFacing, Double> needed = new HashMap<>();
```

Comment:
I'd use `EnumMap` for `wrappers` and `it.unimi.dsi.fastutil.objects.Object2DoubleArrayMap#Object2DoubleArrayMap(6)` for needed/given

Reply:
Tho the double might not work after generification